File: University/lib/Dto/Zystem/ZAPPDto.dart

class ZAPPDto
{
  String ZACONO = "";
  String ZABRNO = "";
  String ZAAPNO = "";
  String ZAAPNA = "";
  String ZAAURL = "";
  String ZAIURL = "";
  String ZAACLR = "";
  double ZAAPSQ = 0;
  String ZAREMA = "";
  String ZASYST = "";
  String ZASTAT = "";
  double ZARCST = 0;
  double ZACRDT = 0;
  double ZACRTM = 0;
  String ZACRUS = "";
  double ZACHDT = 0;
  double ZACHTM = 0;
  String ZACHUS = "";

  ZAPPDto({
    this.ZACONO = "",
    this.ZABRNO = "",
    this.ZAAPNO = "",
    this.ZAAPNA = "",
    this.ZAAURL = "",
    this.ZAIURL = "",
    this.ZAACLR = "",
    this.ZAAPSQ = 0,
    this.ZAREMA = "",
    this.ZASYST = "",
    this.ZASTAT = "",
    this.ZARCST = 0,
    this.ZACRDT = 0,
    this.ZACRTM = 0,
    this.ZACRUS = "",
    this.ZACHDT = 0,
    this.ZACHTM = 0,
    this.ZACHUS = "",
  });

  factory ZAPPDto.fromJson(Map<String, dynamic> json) {
    return ZAPPDto(
      ZACONO: json["ZACONO"] ?? "",
      ZABRNO: json["ZABRNO"] ?? "",
      ZAAPNO: json["ZAAPNO"] ?? "",
      ZAAPNA: json["ZAAPNA"] ?? "",
      ZAAURL: json["ZAAURL"] ?? "",
      ZAIURL: json["ZAIURL"] ?? "",
      ZAACLR: json["ZAACLR"] ?? "",
      ZAAPSQ: json["ZAAPSQ"] ?? 0,
      ZAREMA: json["ZAREMA"] ?? "",
      ZASYST: json["ZASYST"] ?? "",
      ZASTAT: json["ZASTAT"] ?? "",
      ZARCST: json["ZARCST"] ?? 0,
      ZACRDT: json["ZACRDT"] ?? 0,
      ZACRTM: json["ZACRTM"] ?? 0,
      ZACRUS: json["ZACRUS"] ?? "",
      ZACHDT: json["ZACHDT"] ?? 0,
      ZACHTM: json["ZACHTM"] ?? 0,
      ZACHUS: json["ZACHUS"] ?? "",
    );
  }

  Map toMap()
  {
    var map = <String, dynamic>{};
    map["ZACONO"] = ZACONO;
    map["ZABRNO"] = ZABRNO;
    map["ZAAPNO"] = ZAAPNO;
    map["ZAAPNA"] = ZAAPNA;
    map["ZAAURL"] = ZAAURL;
    map["ZAIURL"] = ZAIURL;
    map["ZAACLR"] = ZAACLR;
    map["ZAAPSQ"] = ZAAPSQ;
    map["ZAREMA"] = ZAREMA;
    map["ZASYST"] = ZASYST;
    map["ZASTAT"] = ZASTAT;
    map["ZARCST"] = ZARCST;
    map["ZACRDT"] = ZACRDT;
    map["ZACRTM"] = ZACRTM;
    map["ZACRUS"] = ZACRUS;
    map["ZACHDT"] = ZACHDT;
    map["ZACHTM"] = ZACHTM;
    map["ZACHUS"] = ZACHUS;
    return map;
  }
}

File: University.Service/University.Dto/Zystem/ZAPPDto.cs
using System;
using System.Data;
using System.Collections.Generic;
using System.Runtime.Serialization;

using University.Dto.Base;

namespace University.Dto.Zystem
{
    [DataContract]
    public class ZAPPDto
    {
        #region General Property

        [DataMember]
        public string ZACONO { get; set; }

        [DataMember]
        public string ZABRNO { get; set; }

        [DataMember]
        public string ZAAPNO { get; set; }

        [DataMember]
        public string ZAAPNA { get; set; }

        [DataMember]
        public string ZAAURL { get; set; }

        [DataMember]
        public string ZAIURL { get; set; }

        [DataMember]
        public string ZAACLR { get; set; }

        [DataMember]
        public decimal ZAAPSQ { get; set; }

        [DataMember]
        public string ZAREMA { get; set; }

        [DataMember]
        public string ZASYST { get; set; }

        [DataMember]
        public string ZASTAT { get; set; }

        [DataMember]
        public decimal ZARCST { get; set; }

        [DataMember]
        public decimal ZACRDT { get; set; }

        [DataMember]
        public decimal ZACRTM { get; set; }

        [DataMember]
        public string ZACRUS { get; set; }

        [DataMember]
        public decimal ZACHDT { get; set; }

        [DataMember]
        public decimal ZACHTM { get; set; }

        [DataMember]
        public string ZACHUS { get; set; }

        #endregion

        #region Additional Property

        [DataMember]
        public string ZHUSNO { get; set; }

        [DataMember]
        public string ZAAPNOFr { get; set; }

        [DataMember]
        public string ZAAPNOTo { get; set; }

        //ZHUSNO
        #endregion
    }

    public class ZAPPMappingDto : Mapper<ZAPPDto>
    {
        protected override ZAPPDto PopulateItem(IDataRecord dr)
        {
            ZAPPDto dto = new ZAPPDto();
            for (int i = 0; i < dr.FieldCount; i++)
            {
                MapProperty(dto, dr.GetName(i), dr[i]);
            }
            return dto;
        }
    }
}
